rrdp_xenpm.ml

```diff
@@ -87,8 +87,8 @@ let _ =
 	initialise ();
 	D.warn "Found %d pCPUs" !nr_cpu;
 	if !nr_cpu = 0 then exit 1;
-	(* Share one page per CPU. *)
-	let shared_page_count = !nr_cpu in
+	(* Share two pages per CPU. Consider C-state and P-state CPU datasources *)
+	let shared_page_count = 2 * !nr_cpu in
```

Comment:
This is still a bit of a fiddle. From my reading of the code we can have many C-state and many P-state datasources per CPU. Can we work out how many datasources fit in a page and then work out how many datasources we will have by using `generate_dss` at start of day for the plugin. Then we can set the `shared_page_count` to `(List.length(generate_dss) / dss_per_page) +1`.